src/app/main/service/habit/habit.service.ts

```diff
@@ -46,6 +46,10 @@ export class HabitService {
     return this.http.get<HabitListInterface>(`${habitLink}?lang=${this.language}&page=${page}&size=${size}`);
   }
 
+  getMyAllHabits(page: number, size: number): Observable<HabitListInterface> {
```

Comment:
The `getAllHabits` and `getMyAllHabits` methods look similar 
You can change the method signatures so that they accept the endpoint :
```
getAllHabits(page: number, size: number, endpoint: string): Observable<HabitListInterface> {
    return this.http.get<HabitListInterface>(`${habitLink}${endpoint}?lang=${this.language}&page=${page}&size=${size}`);
  }
  ```